tfx/examples/cifar10_pipeline/cifar10_pipeline_kubeflow.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2019 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""CIFAR-10 example using TFX DSL on Kubeflow."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+from typing import Text
+from tfx.components.evaluator.component import Evaluator
+from tfx.components.example_gen.import_example_gen.component import ImportExampleGen
+from tfx.components.example_validator.component import ExampleValidator
+from tfx.components.model_validator.component import ModelValidator
+from tfx.components.pusher.component import Pusher
+from tfx.components.schema_gen.component import SchemaGen
+from tfx.components.statistics_gen.component import StatisticsGen
+from tfx.components.trainer.component import Trainer
+from tfx.components.transform.component import Transform
+from tfx.orchestration import pipeline
+from tfx.orchestration.kubeflow.runner import KubeflowRunner
+from tfx.orchestration.kubeflow.kubeflow_dag_runner import KubeflowDagRunnerConfig
+from tfx.proto import evaluator_pb2
+from tfx.proto import example_gen_pb2
+from tfx.proto import pusher_pb2
+from tfx.proto import trainer_pb2
+from tfx.utils.dsl_utils import external_input
+from kfp import onprem
+
+
+_pipeline_name = 'cifar10'
+_tfx_root = '/mnt/tfx-pv'
+_pipeline_root = os.path.join(_tfx_root, _pipeline_name)
+_module_file = os.path.join(_pipeline_root, 'cifar10_utils.py')
+_serving_model_dir = os.path.join(_pipeline_root, 'serving_model', _pipeline_name)
+
+
+def _create_pipeline(pipeline_name: Text, pipeline_root: Text, data_root: Text,
+                     module_file: Text, serving_model_dir: Text) -> pipeline.Pipeline:
+    examples = external_input(data_root)
+    input_split = example_gen_pb2.Input(splits=[
+        example_gen_pb2.Input.Split(name='train', pattern='train.tfrecord'),
+        example_gen_pb2.Input.Split(name='eval', pattern='test.tfrecord')
+    ])
+    example_gen = ImportExampleGen(input_base=examples, input_config=input_split)
+    statistics_gen = StatisticsGen(input_data=example_gen.outputs.examples)
+    infer_schema = SchemaGen(stats=statistics_gen.outputs.output)
```

Comment:
You can use infer_feature_shape=True so that you don't need to deal with VarlenFeature in Transform.